app/domain/authentication/security.rb

```diff
@@ -60,7 +62,11 @@ def validate_user_has_access(req)
 
       # Ensure user has access to the service
       has_access = user_role.allowed_to?('authenticate', webservice_resource)
-      raise NotAuthorizedInConjur, req.user_id unless has_access
+      unless has_access
+        Rails.logger.debug("[OIDC] User '#{req.user_id}' is not authorized to " \
```

Comment:
I think the other CodeClimate complaint can be addressed by sticking a helper var `user_id = req.user_id` at the top of this method and using that wherever necessary. See if that fixes it.

Reply:
also here i wouldn;t take CC's advice. 
`req.user_id` is fine and adding the `user_id` method is redundant IMO

Reply:
I think this one is actually somewhat valid. We refer to `req.user_id` four times in the method. Might as well stick it in a variable.

Reply:
ok

Reply:
so i started to fix this but it will be much easier once `security.rb` is a command class. I'll fix this issue in that effort.